plugins/org.jboss.tools.openshift.ui/src/org/jboss/tools/openshift/internal/ui/wizard/deployimage/xpl/ImageName.java

```diff
@@ -0,0 +1,198 @@
+ * limitations under the License.
+ */
+package org.jboss.tools.openshift.internal.ui.wizard.deployimage.xpl;
+
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
+
+/**
+ * Copied from https://github.com/rhuss/docker-maven-plugin/blob/6295761106bbe395d90aa7a20c1f314855c0ad23/src/main/java/org/jolokia/docker/maven/util/ImageName.java
+ * 
+ * Helper class for parsing docker repository/image names:
+ *
+ * <ul>
+ *     <li>If the first part before the slash contains a "." or a ":" it is considered to be a registry URL</li>
+ *     <li>A last part starting with with a ":" is considered to be a tag</li>
+ *     <li>The rest is considered the repository name (which might be separated via slashes)</li>
+ * </ul>
+ *
+ * Example of valid names:
+ *
+ * <ul>
+ *     <li>consol/tomcat-8.0</li>
+ *     <li>consol/tomcat-8.0:8.0.9</li>
+ *     <li>docker.consol.de:5000/tomcat-8.0</li>
+ *     <li>docker.consol.de:5000/jolokia/tomcat-8.0:8.0.9</li>
+ * </ul>
+ *
+ * @author roland
+ * @since 22.07.14
+ */
+public class ImageName {
+
+    // The repository part of the full image
+    private String repository;
+
+    // Registry
+    private String registry;
+
+    // Tag name
+    private String tag;
+
+    // User name
+    private String user;
+
+    /**
+     * Create an image name
+     *
+     * @param fullName The fullname of the image in Docker format.
+     */
+    public ImageName(String fullName) {
+        this(fullName,null);
+    }
+
+    /**
+     * Create an image name with a tag. If a tag is provided (i.e. is not null) then this tag is used.
+     * Otherwise the tag of the provided name is used (if any).
+     *
+     * @param fullName The fullname of the image in Docker format. I
+     * @param givenTag tag to use. Can be null in which case the tag specified in fullName is used.
+     */
+    public ImageName(String fullName,String givenTag) {
+        if (fullName == null) {
+            throw new NullPointerException("Image name must not be null");
+        }
+        Pattern tagPattern = Pattern.compile("^(.+?)(?::([^:/]+))?$");
+        Matcher matcher = tagPattern.matcher(fullName);
+        if (!matcher.matches()) {
+            throw new IllegalArgumentException(fullName + " is not a proper image name ([registry/][repo][:port]");
+        }
+        tag = givenTag != null ?
+                givenTag :
+                matcher.groupCount() > 1 ? matcher.group(2) : null;
+        String rest = matcher.group(1);
+
+        String[] parts = rest.split("\\s*/\\s*");
+        if (parts.length == 1) {
+            registry = null;
+            user = null;
+            repository = parts[0];
+        } else if (parts.length >= 2) {
+            if (isRegistry(parts[0])) {
+                registry = parts[0];
+                user = parts[1];
+                repository = joinTail(parts);
+            } else {
+                registry = null;
+                user = parts[0];
+                repository = rest;
```

Comment:
Inconsistent treatment of white spaces. Other cases dropped them, this case uses 'rest' with them.